server/controllers/socket.controller.ts

```diff
@@ -235,6 +236,52 @@ async function handleOptionSelected(
 	sendRoomData(updatedRoomData);
 }
 
+/**
+ * Updates moderator to existing voter.  Adds former moderator to list of voters.
+ * @param roomCode 4-character code for the room
+ * @param data ModeratorChangeEvent data including new moderatorId
+ * @returns void
+ */
+async function handleModeratorChange(
+	roomCode: string | null,
+	data: ModeratorChangeEvent,
+): Promise<void> {
+	if (!roomCode) throw new Error("Unable to start voting due to no room code.");
+	const roomData = await lookupRoom(roomCode);
+	if (!roomData) return;
+
+	const votersWithoutNewModerator = roomData.voters.filter(
+		(voter) => voter.id !== data.newModeratorId,
+	);
+	const newVoter = {
+		id: roomData.moderator.id,
+		name: roomData.moderator.name,
+		confidence: null,
+		selection: null,
+	};
+	const updatedVoters = [...votersWithoutNewModerator, newVoter];
+
+	const newModeratorName = roomData.voters.find(
```

Comment:
Are we handling the case when the `.find(...)` returns undefined? It shouldn't ever happen, but shouldn't we still handle it?

Reply:
Added more thorough checks.